src/dhali/transaction_utils.py

```diff
@@ -555,3 +555,21 @@ async def validate_estimated_claim(
     )
 
     return uuid_estimate
+
+@firestore.transactional
+def move_document_in_transaction(transaction, source_ref, target_ref):
+    try:
+        source_doc = next(transaction.get(source_ref))
+        if source_doc.exists:
+            data = source_doc.to_dict()
+            transaction.delete(source_ref)
+            transaction.set(target_ref, data)
```

Comment:
Do we not want to swap these around, and check that the setting was successful before deleting from the first collection?